File: java/src/test/java/problems1_100/ReverseIntegerTest.java
package problems1_100;

import org.junit.Test;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ReverseIntegerTest extends SampledTest {
    @Test
    public void test() {
        for (ReverseInteger solution : ReverseInteger.values()) {
            for (ReverseIntegerSample sample : getSamples(ReverseIntegerSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.reverse(sample.input.x))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/ReverseIntegerTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ReverseIntegerTest : SampledTest() {
    @Test
    fun test() =
        ReverseInteger.entries.forEach { solution ->
            getSamples<Array<ReverseIntegerSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.reverse(input.x))
                    .isEqualTo(output)
            }
        }
}
